From 90072fc9a1077491b025d9d81aa606d24bf460fb Mon Sep 17 00:00:00 2001
From: Adithya R <gh0strider.2k18.reborn@gmail.com>
Date: Tue, 3 Aug 2021 20:27:06 +0530
Subject: [PATCH] libcameraservice: Make system cameras available to all
 camera apps

 * big brained realmeme decided to move aux cams to "system-only cameras"
   utilizing the new permission introduced in android 11, thereby breaking
   aux cams in 3rd party camera apps

 * lets avoid this and make system camera accessible to all camera apps;
   legacy apps still wont break because of the aux cams check in fwb

Test: manual, aux cameras accessible by gcam on realme X3
Change-Id: I5db53ffe91a8c28972f1c58bd228cb0f79d7183a
Signed-off-by: Adithya R <gh0strider.2k18.reborn@gmail.com>
Signed-off-by: dinhsan2000 <dinhsan200@gmail.com>
---
 services/camera/libcameraservice/CameraService.cpp | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/services/camera/libcameraservice/CameraService.cpp b/services/camera/libcameraservice/CameraService.cpp
index 0d008c4d1d..b06f579889 100644
--- a/services/camera/libcameraservice/CameraService.cpp
+++ b/services/camera/libcameraservice/CameraService.cpp
@@ -657,9 +657,8 @@ void CameraService::onTorchStatusChangedLocked(const String8& cameraId,
 
 static bool hasPermissionsForSystemCamera(int callingPid, int callingUid,
         bool logPermissionFailure = false) {
-    return checkPermission(sSystemCameraPermission, callingPid, callingUid,
-            logPermissionFailure) &&
-            checkPermission(sCameraPermission, callingPid, callingUid);
+    return checkPermission(sCameraPermission, callingPid, callingUid,
+            logPermissionFailure);
 }
 
 Status CameraService::getNumberOfCameras(int32_t type, int32_t* numCameras) {
